SU2_CFD/src/numerics/NEMO/NEMO_sources.cpp

```diff
@@ -90,146 +86,36 @@ CNumerics::ResidualType<> CSource_NEMO::ComputeChemistry(const CConfig *config)
   for (iVar = 0; iVar < nVar; iVar++) {
     residual[iVar] = 0.0;
   }
+  if (implicit)
+    for (iVar = 0; iVar < nVar; iVar++)
+      for (jVar = 0; jVar < nVar; jVar++)
+        jacobian[iVar][jVar] = 0.0;
 
   /*--- Rename for convenience ---*/
-  T       = V_i[T_INDEX];
-  Tve     = V_i[TVE_INDEX];
+  T   = V_i[T_INDEX];
+  Tve = V_i[TVE_INDEX];
   for(iSpecies = 0; iSpecies < nSpecies; iSpecies++)
     rhos[iSpecies]=V_i[RHOS_INDEX+iSpecies];
 
   /*--- Set mixture state ---*/
   fluidmodel->SetTDStateRhosTTv(rhos, T, Tve);
 
-  ws = fluidmodel->ComputeNetProductionRates();
+  /*---Compute Prodcution/destruction terms ---*/
+  ws = fluidmodel->ComputeNetProductionRates(implicit, V_i, eve_i, Cvve_i,
+                                             dTdU_i, dTvedU_i, jacobian);
 
   for (iSpecies = 0; iSpecies < nSpecies; iSpecies++)
     residual[iSpecies] = ws[iSpecies] *Volume;
 
-  //if (implicit) {
```

Comment:
This chunk of code was moved to ChemistryJacobian in CSU2TCLib.cpp to keep files smaller.